### PR TITLE
feature: add support for testing Pages Functions

### PR DESCRIPTION
**Pull request by petebacondarwin**
## What this PR solves / how to test

3rd part of landing #5508 

Fixes https://github.com/cloudflare/workers-sdk/issues/5288

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required / Maybe required
  - [x] Not required because: covered by fixtures
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [x] Not necessary because: the docs already claim that Pages is supported: See the bottom fo this page https://developers.cloudflare.com/workers/testing/unit-testing/

### CODE REVIEW

File: fixtures/vitest-pool-workers-examples/misc/vitest.config.ts [hunk 1]
```diff
@@ -11,9 +11,15 @@ export default defineWorkersProject({
...
 				miniflare: {
+					kvNamespaces: ["KV_NAMESPACE"],
 					outboundService(request) {
 						return new Response(`fallthrough:${request.method} ${request.url}`);
 					},
+					serviceBindings: {
+						ASSETS(request) {
+							return new Response(`assets:${request.method} ${request.url}`);
+						},
+					},
```
Comment (penalosa) [inline fixtures/vitest-pool-workers-examples/misc/vitest.config.ts:22]: How do we expect people to implement this? Is it worth calling out https://github.com/cloudflare/workers-sdk/blob/8ab8187f0426e1ea4ac8aa63bfde239b51347e74/packages/wrangler/src/miniflare-cli/assets.ts#L28  (what Wrangler uses to generate this) here?

**Review by penalosa — APPROVED** [review]

File: fixtures/vitest-pool-workers-examples/misc/vitest.config.ts (hunk 1 above)
Comment (petebacondarwin) [reply fixtures/vitest-pool-workers-examples/misc/vitest.config.ts:22]: This is really a set of low level tests that are not exactly examples of how to use the integration.
The better example for users is the `pages-functions-unit-integration-self` fixture below.
That uses the `buildPagesASSETSBinding()` helper from `@cloudflare/vitest-pool-workers/config`.

Comment (petebacondarwin) [reply fixtures/vitest-pool-workers-examples/misc/vitest.config.ts:22]: That being said, that helper method is not documented at all in this PR.
I'll create a follow-up ticket to add docs for that.

Comment (petebacondarwin) [reply fixtures/vitest-pool-workers-examples/misc/vitest.config.ts:22]: #5909

**Pull request merged**